src/ir/auth_logic/souffle_emitter.h

```diff
@@ -74,29 +74,29 @@ class SouffleEmitter {
   }
 
   std::string EmitAssertionInner(
-      const datalog::DLIRCondAssertion& cond_assertion) {
+      const datalog::Rule& cond_assertion) {
     std::vector rhs_translated = utils::MapIter<std::string>(
         cond_assertion.rhs(),
         [this](const datalog::Predicate& arg) { return EmitPredicate(arg); });
     return absl::StrCat(EmitPredicate(cond_assertion.lhs()), " :- ",
                         absl::StrJoin(rhs_translated, ", "), ".");
   }
 
-  std::string EmitAssertion(const datalog::DLIRAssertion& assertion) {
-    return std::visit([this](auto value) { return EmitAssertionInner(value); },
-                      assertion.GetValue());
+  std::string EmitAssertion(const datalog::Rule& assertion) {
+    if (assertion.rhs().empty()) return EmitAssertionInner(assertion.lhs()); 
+    return EmitAssertionInner(assertion);
   }
```

Comment:
I'm not sure that the split between `EmitAssertion` and `EmitAssertionInner` serves readability here. Instead of having the `EmitAssertionInner`, imagine that we had a single `EmitAssertion` function that looked like this:

```C++
  std::string EmitAssertionInner(
      const datalog::Rule& cond_assertion) {
    std::vector rhs_translated = utils::MapIter<std::string>(
        cond_assertion.rhs(),
        [this](const datalog::Predicate& arg) { return EmitPredicate(arg); });
    return absl::StrCat(EmitPredicate(cond_assertion.lhs()), (rhs_translated.size() == 0) ? "" : absl::StrCat(" :- ",
                        absl::StrJoin(rhs_translated, ", ")), ".");
  }